File: 11. ContactManager2App/app/src/main/java/com/example/contactsmanager2/MainActivity.java
package com.example.contactsmanager2;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Context;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import com.example.contactsmanager2.databinding.ActivityMainBinding;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    ContactDataAdapter contactDataAdapter;
    ArrayList<Contact> contactArrayList;




    //Binding
    MainActivityClickHandlers mainActivityClickHandlers;
    ActivityMainBinding activityMainBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //Data Binding
        activityMainBinding  = DataBindingUtil.setContentView(this,R.layout.activity_main);
        mainActivityClickHandlers = new MainActivityClickHandlers(this);

        activityMainBinding.setClickHandler(mainActivityClickHandlers);


        //Recycle View

        RecyclerView recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.setHasFixedSize(true);



        contactDataAdapter = new ContactDataAdapter(contactArrayList,this);
        recyclerView.setAdapter(contactDataAdapter);




    }



    public class MainActivityClickHandlers{
        Context context;

        public MainActivityClickHandlers(Context context) {
            this.context = context;
        }


        public void onFABClicked(View view){

            Toast.makeText(context, "oho ji", Toast.LENGTH_SHORT).show();
        }
    }
}

File: __Kotlin Apps/NoteTakingApp/app/src/main/java/com/gurulabs/notificationsender/MainActivity.kt
package com.gurulabs.notificationsender

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.ViewModelProvider
import com.gurulabs.notificationsender.database.NoteDatabase
import com.gurulabs.notificationsender.databinding.ActivityMainBinding
import com.gurulabs.notificationsender.respository.NoteRepository
import com.gurulabs.notificationsender.viewmodel.NoteViewModel
import com.gurulabs.notificationsender.viewmodel.NoteViewModelFactory

class MainActivity : AppCompatActivity() {

    lateinit var noteViewModel: NoteViewModel
    lateinit var binding:ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        setUpViewModel()

    }

    private fun setUpViewModel(){
        val noteRepository = NoteRepository(NoteDatabase(this))
        val viewModelProviderFactory = NoteViewModelFactory(application,noteRepository)
        noteViewModel = ViewModelProvider(this,viewModelProviderFactory).get(NoteViewModel::class.java)


    }
}